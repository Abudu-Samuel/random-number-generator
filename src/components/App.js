import React, { Component } from "react";
import RandomNumber from "./RandomNumber";

class App extends Component {
  state = {
    number: null,
    genNum: [],
    min: 0,
    max: 0
  };

  handleEvent = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  generateRandomNumber = () => {
    let i;
    let gen = [];
    if (this.state.number > 0 && this.state.number <= 5000) {
      for (i = 0; i < this.state.number; i++) {
        const num = Math.floor(100000 + Math.random() * 900000);
        gen.push(`080${num}`);
      }
      this.setState({ genNum: gen });
    }
    return <p>Sorry,minimum is 1 and maximum is 5000</p>;
  };

  sortGeneratedNumber = sortValue => {
    // window.event.preventDefault();
    if (sortValue === "asc") {
      const sort = this.state.genNum.sort((a, b) => Number(a) - Number(b));
      this.setState({ genNum: sort });
    } else {
      const sort = this.state.genNum.sort((a, b) => Number(b) - Number(a));
      this.setState({ genNum: sort });
    }
  };

  minAndMaxValue = num => {
    // window.event.preventDefault();
    const sorted = this.state.genNum.sort((a, b) => Number(a) - Number(b));
    if (num === "min") {
      const min = sorted[0];
      this.setState({ min });
    } else {
      const max = sorted[this.state.genNum.length - 1];
      this.setState({ max });
    }
  };

  render() {
    return (
      <div className="container-fluid">
        <div className="row mt-3">
          <div className="col-md-6 offset-md-4 mb-3">
            <h3>Random number generator</h3>
          </div>
          <div className="col-md-5 offset-md-3">
            <RandomNumber
              handleEvent={this.handleEvent}
              generateRandomNumber={this.generateRandomNumber}
              genNum={this.state.genNum}
              sortGeneratedNumber={this.sortGeneratedNumber}
              minAndMaxValue={this.minAndMaxValue}
              min={this.state.min}
              max={this.state.max}
              number={this.state.number}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
