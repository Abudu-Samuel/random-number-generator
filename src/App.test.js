import React from "react";
import { render, fireEvent } from "react-testing-library";
import App from "./components/App";

describe("App test", () => {
  const wrapper = render(<App />);
  const { getByText, getByTestId } = wrapper;

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("generates number when the generate number button is clicked", () => {
    const numberValue = getByTestId("number");
    fireEvent.change(numberValue, { target: { value: 5 } });
    const generateRandomNumberButton = getByText("Generate Number");
    const number = getByTestId("number");
    fireEvent.click(generateRandomNumberButton);

    expect(number).toBeTruthy();
  });

  it("shows the maximum number when the max button is clicked", () => {
    const generateRandomNumberButton = getByText("Generate Number");
    fireEvent.click(generateRandomNumberButton);
    const maxNumberButton = getByText("Max");
    const number = getByTestId("number");
    fireEvent.click(maxNumberButton);

    expect(number).toBeTruthy();
  });

  it("shows the minimum number when the min button is clicked", () => {
    const generateRandomNumberButton = getByText("Generate Number");
    fireEvent.click(generateRandomNumberButton);
    const minNumberButton = getByText("Min");
    const number = getByTestId("number");
    fireEvent.click(minNumberButton);

    expect(number).toBeTruthy();
  });

  it("sort numbers when the ascending button is clicked", () => {
    const generateRandomNumberButton = getByText("Generate Number");
    fireEvent.click(generateRandomNumberButton);
    const ascNumberButton = getByText("Ascending");
    const number = getByTestId("number");
    fireEvent.click(ascNumberButton);

    expect(number).toBeTruthy();
  });

  it("sort numbers when the descending button is clicked", () => {
    const generateRandomNumberButton = getByText("Generate Number");
    fireEvent.click(generateRandomNumberButton);
    const dscNumberButton = getByText("Descending");
    const number = getByTestId("number");
    fireEvent.click(dscNumberButton);

    expect(number).toBeTruthy();
  });
});
