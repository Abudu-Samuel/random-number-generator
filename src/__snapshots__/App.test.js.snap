// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid"
      >
        <div
          class="row mt-3"
        >
          <div
            class="col-md-6 offset-md-4 mb-3"
          >
            <h3>
              Random number generator
            </h3>
          </div>
          <div
            class="col-md-5 offset-md-3"
          >
            <div>
              <div
                class="input-group mb-3 "
              >
                <input
                  class="form-control"
                  data-testid="number"
                  name="number"
                  placeholder="Enter number"
                  type="number"
                />
                <div
                  class="input-group-append"
                >
                  <button
                    class="btn btn-outline-primary"
                    type="button"
                  >
                    Generate Number
                  </button>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="form-inline offset-md-2"
                >
                  <div
                    class="custom-control custom-checkbox"
                  >
                    <button
                      class="btn btn-sm btn-outline-primary"
                      data-testid="number"
                    >
                      Ascending
                    </button>
                  </div>
                  <div
                    class="custom-control custom-checkbox"
                  >
                    <button
                      class="btn btn-sm btn-outline-primary"
                    >
                      Descending
                    </button>
                  </div>
                  <div
                    class="custom-control custom-checkbox"
                  >
                    <button
                      class="btn btn-sm btn-outline-primary"
                      data-target="#exampleModal"
                      data-testid="number"
                      data-toggle="modal"
                    >
                      Min
                    </button>
                  </div>
                  <div
                    aria-hidden="true"
                    aria-labelledby="exampleModalLabel"
                    class="modal fade"
                    id="exampleModal"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="modal-dialog"
                      role="document"
                    >
                      <div
                        class="modal-content"
                      >
                        <div
                          class="modal-header"
                        >
                          <h5
                            class="modal-title text-center"
                            id="exampleModalLabel"
                          >
                            Minimum Number
                          </h5>
                          <button
                            aria-label="Close"
                            class="close"
                            data-dismiss="modal"
                            type="button"
                          >
                            <span
                              aria-hidden="true"
                            >
                              ×
                            </span>
                          </button>
                        </div>
                        <div
                          class="modal-body"
                        >
                          Minimum Number generated = 
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="custom-control custom-checkbox mr-5"
                  >
                    <button
                      class="btn btn-sm btn-outline-primary"
                      data-target="#exampleModal2"
                      data-testid="number"
                      data-toggle="modal"
                    >
                      Max
                    </button>
                  </div>
                  <div
                    aria-hidden="true"
                    aria-labelledby="exampleModalLabel"
                    class="modal fade"
                    id="exampleModal2"
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="modal-dialog"
                      role="document"
                    >
                      <div
                        class="modal-content"
                      >
                        <div
                          class="modal-header"
                        >
                          <h5
                            class="modal-title text-center"
                            id="exampleModalLabel"
                          >
                            Maximum Number
                          </h5>
                          <button
                            aria-label="Close"
                            class="close"
                            data-dismiss="modal"
                            type="button"
                          >
                            <span
                              aria-hidden="true"
                            >
                              ×
                            </span>
                          </button>
                        </div>
                        <div
                          class="modal-body"
                        >
                          Maximum Number generated = 
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="custom-control custom-checkbox mt-1 offset-md-3"
                  >
                    <a
                      download="numbers.txt"
                      href="data:text/plain;charset=utf-8,"
                    >
                      <button
                        class="btn btn-sm btn-outline-primary"
                      >
                        Dowload numbers
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="row mt-4"
                data-testid="number-list-section"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid"
    >
      <div
        class="row mt-3"
      >
        <div
          class="col-md-6 offset-md-4 mb-3"
        >
          <h3>
            Random number generator
          </h3>
        </div>
        <div
          class="col-md-5 offset-md-3"
        >
          <div>
            <div
              class="input-group mb-3 "
            >
              <input
                class="form-control"
                data-testid="number"
                name="number"
                placeholder="Enter number"
                type="number"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-outline-primary"
                  type="button"
                >
                  Generate Number
                </button>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="form-inline offset-md-2"
              >
                <div
                  class="custom-control custom-checkbox"
                >
                  <button
                    class="btn btn-sm btn-outline-primary"
                    data-testid="number"
                  >
                    Ascending
                  </button>
                </div>
                <div
                  class="custom-control custom-checkbox"
                >
                  <button
                    class="btn btn-sm btn-outline-primary"
                  >
                    Descending
                  </button>
                </div>
                <div
                  class="custom-control custom-checkbox"
                >
                  <button
                    class="btn btn-sm btn-outline-primary"
                    data-target="#exampleModal"
                    data-testid="number"
                    data-toggle="modal"
                  >
                    Min
                  </button>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="exampleModalLabel"
                  class="modal fade"
                  id="exampleModal"
                  role="dialog"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <h5
                          class="modal-title text-center"
                          id="exampleModalLabel"
                        >
                          Minimum Number
                        </h5>
                        <button
                          aria-label="Close"
                          class="close"
                          data-dismiss="modal"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        Minimum Number generated = 
                        0
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="custom-control custom-checkbox mr-5"
                >
                  <button
                    class="btn btn-sm btn-outline-primary"
                    data-target="#exampleModal2"
                    data-testid="number"
                    data-toggle="modal"
                  >
                    Max
                  </button>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="exampleModalLabel"
                  class="modal fade"
                  id="exampleModal2"
                  role="dialog"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <h5
                          class="modal-title text-center"
                          id="exampleModalLabel"
                        >
                          Maximum Number
                        </h5>
                        <button
                          aria-label="Close"
                          class="close"
                          data-dismiss="modal"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        Maximum Number generated = 
                        0
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="custom-control custom-checkbox mt-1 offset-md-3"
                >
                  <a
                    download="numbers.txt"
                    href="data:text/plain;charset=utf-8,"
                  >
                    <button
                      class="btn btn-sm btn-outline-primary"
                    >
                      Dowload numbers
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="row mt-4"
              data-testid="number-list-section"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
